File: boj/10845.cpp
#include <iostream>
#include <queue>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int n;
    queue<int> q;

    cin >> n;
    for (int i = 0; i < n; i++) {
        string str;
        
        cin >> str;
        if (str == "push") {
            int num;

            cin >> num;
            q.push(num);
        } else if (str == "pop") {
            if (!q.empty()) {
                cout << q.front() << "\n";
                q.pop();
            } else {
                cout << -1 << "\n";
            }
        } else if (str == "size") {
            cout << q.size() << "\n";
        } else if (str == "empty") {
            int num =  q.empty() ? 1 : 0;
            cout << num << "\n";
        } else if (str == "front") {
            if (!q.empty()) {
                cout << q.front() << "\n";
            } else {
                cout << -1 << "\n";
            }
        } else if (str == "back") {
            if (!q.empty()) {
                cout << q.back() << "\n";
            } else {
                cout << -1 << "\n";
            }
        }
    }
}

File: boj/10845.py
import sys

input = sys.stdin.readline
N = int(input())
queue = []
for _ in range(N):
    cmd = input().split()
    if cmd[0] == "push":
        queue.append(cmd[1])
    elif cmd[0] == "pop":
        print(queue.pop(0)) if queue else print(-1)
    elif cmd[0] == "size":
        print(len(queue))
    elif cmd[0] == "empty":
        print(0) if queue else print(1)
    elif cmd[0] == "front":
        print(queue[0]) if queue else print(-1)
    elif cmd[0] == "back":
        print(queue[-1]) if queue else print(-1)
